src/embeds/items.js

```diff
@@ -3,5 +3,14 @@
 import Items from './components/Items';
 
 export default function (element: HTMLElement, ids: Array<number>) {
-  return () => <Items ids={ids} />;
+  const statIds = ids.map((id) => {
+    const rawId = element.getAttribute(`data-armory-${id}-stat`);
+    if (!rawId) {
+      return [];
+    }
+
+    return { [id]: parseInt(rawId, 10) };
+  })
+  .reduce((map, next) => Object.assign(map, next), {});
```

Comment:
Initially i wrote it that way, but compiler breaks on 
```
        formattedErrors = formattedErrors.filter(isLikelyASyntaxError);
                        ^

TypeError: Assignment to constant variable.
```

let's merge it as Object.assign, and you can try fix that, the truth is i'm coming from typescript and not flow + es6, in typescript it just works as you mentioned.. not sure how the toolchain is working here.